presto-spi/src/main/java/io/prestosql/spi/session/PropertyMetadata.java

```diff
@@ -199,4 +203,24 @@ public Object encode(T value)
                 String.class::cast,
                 object -> object);
     }
+
+    public static <T extends Enum<T>> PropertyMetadata<T> enumProperty(String name, String description, Class<T> type, T defaultValue, boolean hidden)
+    {
+        return new PropertyMetadata<>(
+                name,
+                description,
+                createUnboundedVarcharType(),
+                type,
+                defaultValue,
+                hidden,
+                value -> {
+                    try {
+                        return Enum.valueOf(type, ((String) value).toUpperCase(ENGLISH));
+                    }
+                    catch (IllegalArgumentException e) {
+                        throw new PrestoException(GENERIC_USER_ERROR, format("Invalid value [%s]. Valid values: %s", value, EnumSet.allOf(type)), e);
```

Comment:
This should stay as `IllegalArgumentException`. The property manager will convert it into the correct type of error code (for the different property types).